### PR TITLE
Add popN functionality

### PR DESCRIPTION
**Pull request by chrisbianca**
As per issue https://github.com/exponentjs/ex-navigation/issues/12 I have added a popN function which pops multiple screens.

### CODE REVIEW

File: src/ExNavigationActions.js [hunk 1]
```diff
@@ -43,6 +43,14 @@ export default class ExNavigationActions {
     };
   }
 
+  static popN(navigatorUID, n) {
+    return {
+      type: ActionTypes.POP_N,
+      navigatorUID,
+      n
```
Comment (chirag04) [inline src/ExNavigationActions.js]: trailing comma

File: src/ExNavigationReducer.js [hunk 2]
```diff
@@ -145,6 +145,28 @@ class ExNavigationReducer {
...
+
+    if (navigatorState.type === 'slidingTab') {
+      return _updateNavigator(
+        state,
+        navigatorUID,
+        {...navigatorState, index: 0 },
+      );
+    }
+
+    const sliceTo = n > navigatorState.routes.length ? 1 : -n;
```
Comment (chirag04) [inline src/ExNavigationReducer.js]: can this check be an invariant?

File: src/ExNavigationReducer.js [hunk 3]
```diff
@@ -145,6 +145,28 @@ class ExNavigationReducer {
...
+    const routes = navigatorState.routes.slice(0, sliceTo);
```
Comment (chirag04) [inline src/ExNavigationReducer.js]: a lot of logic her is exactly like pop_to_top. Can we share the code here?

Comment (chirag04) [inline src/ExNavigationReducer.js]: I would expect popToTop to just do popN(routes.length)

**Comment by chirag04** [comment]
thanks for the pr @chrisbianca. This is looking great.

**Comment by brentvatne** [comment]
@chrisbianca - what do you think about just using the `pop` function and defaulting it to 1? If the value is greater than 1, then call `Actions.popN`

**Comment by chrisbianca** [comment]
@brentvatne - happy to change it to do that instead if that's what you'd prefer.  I went with popN as this is the same as the RN Navigator, though I'm assuming that is going to be deprecated at some point in the future so probably no need to maintain the convention.

Let me know what you're like me to do...

Also, see @chirag04's comment about using an invariant - would be interested in your thoughts.

**Comment by brentvatne** [comment]
I'd prefer `pop(n)`, smaller public api surface area

File: src/ExNavigationReducer.js (hunk 2 above)
Comment (brentvatne) [inline src/ExNavigationReducer.js]: I think this solution seems fine to me -- I think if you use `pop(5)` but only have 3 routes in the stack, you probably want it to just work with 3, but warn you. So we should add a `console.warn` in there if in dev mode.

**Comment by chrisbianca** [comment]
I've updated to pop(n) and added the warning when in dev mode

**Comment by chrisbianca** [comment]
@brentvatne Any chance of getting this PR merged?

**Comment by brentvatne** [comment]
Thanks @chrisbianca!

**Pull request closed without merging**